lib/experimental/Widgets/Content/ClockIn/ClockInGraph/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { formatTime24Hours } from "@/lib/date"
+import { Cell, Pie, PieChart } from "recharts"
+
+export type ClockInStatus = "clocked-in" | "break" | "clocked-out"
+
+export interface ClockInGraphProps {
+  data?: {
+    from: Date
+    to: Date
+    variant: ClockInStatus
+  }[]
+  remainingMinutes?: number
+}
+
+export const CLOCK_IN_COLORS = {
+  "clocked-in": "hsl(var(--positive-50))",
+  break: "hsl(var(--promote-50))",
+  empty: "hsl(var(--neutral-10))",
+  "clocked-out": "hsl(var(--neutral-10))",
+  overtime: "hsl(var(--warning-50))",
+}
+
+const EMPTY_LABEL = "--:--"
+
+export function ClockInGraph({
+  data = [],
+  remainingMinutes,
+}: ClockInGraphProps) {
+  const clockedInAt = data.find((entry) => entry.variant === "clocked-in")?.from
+  const clockedOutAt = data.find(
+    (entry) => entry.variant === "clocked-out"
+  )?.from
+
+  const lastEntry = data[data.length - 1]
+
+  const lastClockedInEntry = data
+    .slice()
+    .reverse()
+    .find((entry) => entry.variant === "clocked-in")
+
+  const leftEntry = (() => {
+    if (remainingMinutes && remainingMinutes > 0) {
+      return {
+        value: remainingMinutes,
+        color: CLOCK_IN_COLORS.empty,
+      }
+    }
+
+    if (remainingMinutes && remainingMinutes < 0) {
+      return {
+        value: Math.abs(remainingMinutes),
+        color: CLOCK_IN_COLORS.overtime,
+      }
+    }
+
+    return {
+      value: 1,
+      color: CLOCK_IN_COLORS.empty,
+    }
+  })()
```

Comment:
IFEE? This function doesn't modify the state, it should be extracted from the component.